jbmc/src/java_bytecode/java_object_factory_parameters.cpp

```diff
@@ -22,30 +22,21 @@ void java_object_factory_parameterst::set(const optionst &options)
   {
     const auto &interval_string =
       options.get_option("java-assume-inputs-interval");
-    const std::regex limits_regex("\\[(-\\d+|\\d*):(-\\d+|\\d*)\\]");
-    std::smatch base_match;
-    if(!std::regex_match(interval_string, base_match, limits_regex))
+    auto interval = interval_uniont::of_string(interval_string);
+    if(!interval.has_value())
     {
       throw invalid_command_line_argument_exceptiont(
-        "interval must be of the form [int:int], [int:] or [:int]",
+        "argument must be a sequence of intervals of the form [int:int], [int:]"
+        " or [:int]",
         "--java-assume-inputs-interval");
     }
-    assume_inputs_interval = [&]() -> integer_intervalt {
-      integer_intervalt interval;
-      if(!base_match[1].str().empty())
-        interval.make_ge_than(string2integer(base_match[1].str()));
-      if(!base_match[2].str().empty())
-        interval.make_le_than(string2integer(base_match[2].str()));
-      if(interval.is_top())
-        throw invalid_command_line_argument_exceptiont(
-          "at least one of the interval bounds must be given",
-          "--java-assume-inputs-interval");
-      if(interval.empty())
-        throw invalid_command_line_argument_exceptiont(
-          "interval is empty, lower limit cannot be bigger than upper limit",
-          "--java-assume-inputs-interval");
-      return interval;
-    }();
+    if(interval->is_empty())
+    {
+      throw invalid_command_line_argument_exceptiont(
+        "interval is empty, lower limit cannot be bigger than upper limit",
```

Comment:
:grey_question: Is this the only situation when the interval ends up empty? I cannot think of any other but you may have a deeper insight into it.

Reply:
I guess if you give it an empty string, but I'm not sure the option parser would accept that...